cypress-io/cypress
# chore(launchpad): removing ipc from launchpad in order to enable real cy e2e testing

## Context

To test the Launchpad GUI flow, we'll need to use Cypress E2E. This PR removes IPC which allows Cy to visit launchpad like it's a normal website. This minimizes the amount of mocking requires in our e2e tests ⚡.

### How to Test

1. yarn workspace @packages/launchpad dev
2. yarn workspace @packages/launchpad start
3. Open your local browser (Chrome, Firefox, wherever)
4. Visit http://localhost:3001

**Currently:** Nothing renders. An error throws in the console when trying to access `window.ipc`.
**After change:** You should see the launchpad.

Note: The launchpad may be particularly slow to render due to a known issue w/ browser detection.

### Explanation of Architecture

#### Currently
Cy's launchpad equivalent, desktop-gui, requires a lot of stubbing in order to validate its functionality because Cy E2E has no support for testing electron apps that may depend on IPC, etc.

#### Now
In our new launchpad, we use graphql as our API, and graphql can use http as a transport layer, instead of IPC. Because we're using graphql, **we're able to avoid stubbing all IPC calls** which is what desktop-gui currently has to do.

#### After this PR we...
* Should be able to open Cy Launchpad inside of my local browser without stubbing anything.
* Should be unblocked to host Cy Launchpad via express and hit it like it's a normal website via Cy E2E.

## Bonus Wins
Because you're using your normal browser, you can develop Launchpad using the urql, Vue, and other normal devtools! 🎉 

#### Related Resources
- Closes [UNIFY-343](https://cypress-io.atlassian.net/browse/UNIFY-343)
